src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.ts

```diff
@@ -72,4 +104,79 @@ export class ExamChecklistComponent implements OnChanges, OnInit, OnDestroy {
+        if (this.exam.course?.id !== undefined && this.exam.id !== undefined) {
+            this.examManagementService.evaluateQuizExercises(this.exam.course.id, this.exam.id).subscribe({
+                next: (res) => {
+                    this.alertService.success('artemisApp.studentExams.evaluateQuizExerciseSuccess', { number: res?.body });
+                    this.existsUnassessedQuizzes = false;
+                    this.isEvaluatingQuizExercises = false;
+                },
+                error: (err: HttpErrorResponse) => {
+                    this.handleError('artemisApp.studentExams.evaluateQuizExerciseFailure', err);
+                    this.isEvaluatingQuizExercises = false;
+                },
+            });
+        } else {
+            throw new Error(`Cannot evaluate quiz exercises due to missing course or exam id.`);
+        }
+    }
+
+    /**
+     * Evaluates all the unsubmitted Text and Modelling submissions to 0
+     */
+    assessUnsubmittedExamModelingAndTextParticipations() {
+        this.isAssessingUnsubmittedExams = true;
+        if (this.exam.course?.id !== undefined && this.exam.id !== undefined) {
+            this.examManagementService.assessUnsubmittedExamModelingAndTextParticipations(this.exam.course.id, this.exam.id).subscribe({
+                next: (res) => {
+                    this.alertService.success('artemisApp.studentExams.assessUnsubmittedStudentExamsSuccess', { number: res?.body });
+                    this.existsUnsubmittedExercises = false;
+                    this.isAssessingUnsubmittedExams = false;
+                },
+                error: (err: HttpErrorResponse) => {
+                    this.handleError('artemisApp.studentExams.assessUnsubmittedStudentExamsFailure', err);
+                    this.isAssessingUnsubmittedExams = false;
+                },
+            });
+        } else {
+            throw new Error(`Cannot unsubmitted exercises due to missing course or exam id.`);
+        }
+    }
+
+    /**
+     * Shows the translated error message if an error key is available in the error response. Otherwise it defaults to the generic alert.
+     * @param translationString the string identifier in the translation service for the text. This is ignored if the response does not contain an error message or error key.
+     * @param err the error response
+     */
+    private handleError(translationString: string, err: HttpErrorResponse) {
+        let errorDetail;
+        if (err?.error && err.error.errorKey) {
+            errorDetail = this.artemisTranslatePipe.transform(err.error.errorKey);
+        } else {
+            errorDetail = err?.error?.message;
+        }
+        if (errorDetail) {
+            this.alertService.error(translationString, { message: errorDetail });
+        } else {
+            // Sometimes the response does not have an error field, so we default to generic error handling
+            onError(this.alertService, err);
+        }
+    }
```

Comment:
This should not be necessary - The alertService is able to translate messages. You should be able to directly use this.alertService.error('artemisApp...'); in your methods.

Reply:
Thanks so much for the info, I didn't know this because there were different methods in the code :) Changed